src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs

```diff
@@ -407,5 +432,29 @@ private Task ProcessNegotiate(HttpContext context)
 
             return connectionState;
         }
+
+        private IList<Message> ParseSendBatch(ref BytesReader payload, MessageFormat messageFormat)
+        {
+            var messages = new List<Message>();
+
+            if (payload.Unread.Length == 0)
```

Comment:
In general we've tried to be lenient with empty payloads, so I'm just doing the same here. An empty payload here is odd, but not necessarily exceptional in that our behavior is deterministic (unlike a parse error in a payload where we don't know how to recover).